### PR TITLE
Add .NET Core, WindowsDesktop runtime packs

### PR DESCRIPTION
**Pull request by dagood**
Creates runtime packs for .NET Core and Windows Desktop. Fixes a few issues vs. the old runtime packages (see https://github.com/dotnet/core-setup/issues/4775 checklist).

Resolves https://github.com/dotnet/core-setup/issues/4775.

I added `ParallelSafeRestore` because with this change, we are building `runtime.<rid>.Microsoft.NETCore.App` and `Microsoft.NETCore.App.Runtime.<rid>` in parallel, which both restore `netcoreapp.depproj` and trigger the problem described in the code comment. (Filed https://github.com/dotnet/core-setup/issues/6696 to get rid of it.)

---

Also contains a few fixes to make `binlogs` more reliably available in the builds.

I'm including the bundle installer `light.exe` retry functionality in this PR because it seems that producing more assets makes the antivirus lock issue more likely to repro, and the retry is more reliable than only sleeping for only one second. Resolves https://github.com/dotnet/core-setup/issues/6651.

<details><summary>Links to artifacts from a mock official build (click me)</summary>

* [Microsoft.NETCore.App.Runtime.linux-arm.3.0.0-preview7-27804-05.nupkg](https://dagoodbuildoutput.blob.core.windows.net/public-test-bits/runtime-pack/3.0.0-preview7-27804-05/Microsoft.NETCore.App.Runtime.linux-arm.3.0.0-preview7-27804-05.nupkg)
* [Microsoft.NETCore.App.Runtime.linux-arm.3.0.0-preview7-27804-05.symbols.nupkg](https://dagoodbuildoutput.blob.core.windows.net/public-test-bits/runtime-pack/3.0.0-preview7-27804-05/Microsoft.NETCore.App.Runtime.linux-arm.3.0.0-preview7-27804-05.symbols.nupkg)
* [Microsoft.NETCore.App.Runtime.linux-arm64.3.0.0-preview7-27804-05.nupkg](https://dagoodbuildoutput.blob.core.windows.net/public-test-bits/runtime-pack/3.0.0-preview7-27804-05/Microsoft.NETCore.App.Runtime.linux-arm64.3.0.0-preview7-27804-05.nupkg)
* [Microsoft.NETCore.App.Runtime.linux-arm64.3.0.0-preview7-27804-05.symbols.nupkg](https://dagoodbuildoutput.blob.core.windows.net/public-test-bits/runtime-pack/3.0.0-preview7-27804-05/Microsoft.NETCore.App.Runtime.linux-arm64.3.0.0-preview7-27804-05.symbols.nupkg)
* [Microsoft.NETCore.App.Runtime.linux-musl-arm64.3.0.0-preview7-27804-05.nupkg](https://dagoodbuildoutput.blob.core.windows.net/public-test-bits/runtime-pack/3.0.0-preview7-27804-05/Microsoft.NETCore.App.Runtime.linux-musl-arm64.3.0.0-preview7-27804-05.nupkg)
* [Microsoft.NETCore.App.Runtime.linux-musl-arm64.3.0.0-preview7-27804-05.symbols.nupkg](https://dagoodbuildoutput.blob.core.windows.net/public-test-bits/runtime-pack/3.0.0-preview7-27804-05/Microsoft.NETCore.App.Runtime.linux-musl-arm64.3.0.0-preview7-27804-05.symbols.nupkg)
* [Microsoft.NETCore.App.Runtime.linux-musl-x64.3.0.0-preview7-27804-05.nupkg](https://dagoodbuildoutput.blob.core.windows.net/public-test-bits/runtime-pack/3.0.0-preview7-27804-05/Microsoft.NETCore.App.Runtime.linux-musl-x64.3.0.0-preview7-27804-05.nupkg)
* [Microsoft.NETCore.App.Runtime.linux-musl-x64.3.0.0-preview7-27804-05.symbols.nupkg](https://dagoodbuildoutput.blob.core.windows.net/public-test-bits/runtime-pack/3.0.0-preview7-27804-05/Microsoft.NETCore.App.Runtime.linux-musl-x64.3.0.0-preview7-27804-05.symbols.nupkg)
* [Microsoft.NETCore.App.Runtime.linux-x64.3.0.0-preview7-27804-05.nupkg](https://dagoodbuildoutput.blob.core.windows.net/public-test-bits/runtime-pack/3.0.0-preview7-27804-05/Microsoft.NETCore.App.Runtime.linux-x64.3.0.0-preview7-27804-05.nupkg)
* [Microsoft.NETCore.App.Runtime.linux-x64.3.0.0-preview7-27804-05.symbols.nupkg](https://dagoodbuildoutput.blob.core.windows.net/public-test-bits/runtime-pack/3.0.0-preview7-27804-05/Microsoft.NETCore.App.Runtime.linux-x64.3.0.0-preview7-27804-05.symbols.nupkg)
* [Microsoft.NETCore.App.Runtime.osx-x64.3.0.0-preview7-27804-05.nupkg](https://dagoodbuildoutput.blob.core.windows.net/public-test-bits/runtime-pack/3.0.0-preview7-27804-05/Microsoft.NETCore.App.Runtime.osx-x64.3.0.0-preview7-27804-05.nupkg)
* [Microsoft.NETCore.App.Runtime.osx-x64.3.0.0-preview7-27804-05.symbols.nupkg](https://dagoodbuildoutput.blob.core.windows.net/public-test-bits/runtime-pack/3.0.0-preview7-27804-05/Microsoft.NETCore.App.Runtime.osx-x64.3.0.0-preview7-27804-05.symbols.nupkg)
* [Microsoft.NETCore.App.Runtime.rhel.6-x64.3.0.0-preview7-27804-05.nupkg](https://dagoodbuildoutput.blob.core.windows.net/public-test-bits/runtime-pack/3.0.0-preview7-27804-05/Microsoft.NETCore.App.Runtime.rhel.6-x64.3.0.0-preview7-27804-05.nupkg)
* [Microsoft.NETCore.App.Runtime.rhel.6-x64.3.0.0-preview7-27804-05.symbols.nupkg](https://dagoodbuildoutput.blob.core.windows.net/public-test-bits/runtime-pack/3.0.0-preview7-27804-05/Microsoft.NETCore.App.Runtime.rhel.6-x64.3.0.0-preview7-27804-05.symbols.nupkg)
* [Microsoft.NETCore.App.Runtime.win-arm.3.0.0-preview7-27804-05.nupkg](https://dagoodbuildoutput.blob.core.windows.net/public-test-bits/runtime-pack/3.0.0-preview7-27804-05/Microsoft.NETCore.App.Runtime.win-arm.3.0.0-preview7-27804-05.nupkg)
* [Microsoft.NETCore.App.Runtime.win-arm.3.0.0-preview7-27804-05.symbols.nupkg](https://dagoodbuildoutput.blob.core.windows.net/public-test-bits/runtime-pack/3.0.0-preview7-27804-05/Microsoft.NETCore.App.Runtime.win-arm.3.0.0-preview7-27804-05.symbols.nupkg)
* [Microsoft.NETCore.App.Runtime.win-arm64.3.0.0-preview7-27804-05.nupkg](https://dagoodbuildoutput.blob.core.windows.net/public-test-bits/runtime-pack/3.0.0-preview7-27804-05/Microsoft.NETCore.App.Runtime.win-arm64.3.0.0-preview7-27804-05.nupkg)
* [Microsoft.NETCore.App.Runtime.win-arm64.3.0.0-preview7-27804-05.symbols.nupkg](https://dagoodbuildoutput.blob.core.windows.net/public-test-bits/runtime-pack/3.0.0-preview7-27804-05/Microsoft.NETCore.App.Runtime.win-arm64.3.0.0-preview7-27804-05.symbols.nupkg)
* [Microsoft.NETCore.App.Runtime.win-x64.3.0.0-preview7-27804-05.nupkg](https://dagoodbuildoutput.blob.core.windows.net/public-test-bits/runtime-pack/3.0.0-preview7-27804-05/Microsoft.NETCore.App.Runtime.win-x64.3.0.0-preview7-27804-05.nupkg)
* [Microsoft.NETCore.App.Runtime.win-x64.3.0.0-preview7-27804-05.symbols.nupkg](https://dagoodbuildoutput.blob.core.windows.net/public-test-bits/runtime-pack/3.0.0-preview7-27804-05/Microsoft.NETCore.App.Runtime.win-x64.3.0.0-preview7-27804-05.symbols.nupkg)
* [Microsoft.NETCore.App.Runtime.win-x86.3.0.0-preview7-27804-05.nupkg](https://dagoodbuildoutput.blob.core.windows.net/public-test-bits/runtime-pack/3.0.0-preview7-27804-05/Microsoft.NETCore.App.Runtime.win-x86.3.0.0-preview7-27804-05.nupkg)
* [Microsoft.NETCore.App.Runtime.win-x86.3.0.0-preview7-27804-05.symbols.nupkg](https://dagoodbuildoutput.blob.core.windows.net/public-test-bits/runtime-pack/3.0.0-preview7-27804-05/Microsoft.NETCore.App.Runtime.win-x86.3.0.0-preview7-27804-05.symbols.nupkg)
* [Microsoft.WindowsDesktop.App.Runtime.win-x64.3.0.0-preview7-27804-05.nupkg](https://dagoodbuildoutput.blob.core.windows.net/public-test-bits/runtime-pack/3.0.0-preview7-27804-05/Microsoft.WindowsDesktop.App.Runtime.win-x64.3.0.0-preview7-27804-05.nupkg)
* [Microsoft.WindowsDesktop.App.Runtime.win-x64.3.0.0-preview7-27804-05.symbols.nupkg](https://dagoodbuildoutput.blob.core.windows.net/public-test-bits/runtime-pack/3.0.0-preview7-27804-05/Microsoft.WindowsDesktop.App.Runtime.win-x64.3.0.0-preview7-27804-05.symbols.nupkg)
* [Microsoft.WindowsDesktop.App.Runtime.win-x86.3.0.0-preview7-27804-05.nupkg](https://dagoodbuildoutput.blob.core.windows.net/public-test-bits/runtime-pack/3.0.0-preview7-27804-05/Microsoft.WindowsDesktop.App.Runtime.win-x86.3.0.0-preview7-27804-05.nupkg)
* [Microsoft.WindowsDesktop.App.Runtime.win-x86.3.0.0-preview7-27804-05.symbols.nupkg](https://dagoodbuildoutput.blob.core.windows.net/public-test-bits/runtime-pack/3.0.0-preview7-27804-05/Microsoft.WindowsDesktop.App.Runtime.win-x86.3.0.0-preview7-27804-05.symbols.nupkg)

</details>

/cc @nguerrera @dsplaisted @johnbeisner

### CODE REVIEW

File: src/pkg/projects/netcoreapp/src/netcoreapp.depproj [hunk 1]
```diff
@@ -60,7 +60,14 @@
       <_ToolsToPackage Include="$(_runtimePackageDir)tools/**/*.*"/>
       <FilesToPackage Include="@(_ToolsToPackage)">
         <NuGetPackageId>$(_runtimePackageId)</NuGetPackageId>
-        <TargetPath>tools/%(RecursiveDir)</TargetPath>
+        <!--
+          Trim '/' at the end of the target path, if exists. The packaging tools automatically add
```
Comment (ericstj) [inline src/pkg/projects/netcoreapp/src/netcoreapp.depproj]: We should fix that in packaging tools.

File: src/pkg/projects/netcoreapp/pkg/Microsoft.NETCore.App.Runtime.pkgproj [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  <!--
+    Include HostPolicy and HostResolver based on legacy packages. This can be collapsed once the
+    legacy packages are no longer used.
+  -->
+  <Import Project="..\..\Microsoft.NETCore.DotNetHostPolicy\Microsoft.NETCore.DotNetHostPolicy.props" />
+  <Import Project="..\..\Microsoft.NETCore.DotNetHostResolver\Microsoft.NETCore.DotNetHostResolver.props" />
+
+  <!--
+    The shared framework should have createdump for diagnostics, but not self-contained apps. This
```
[the remark this answers is not in the document]
Comment (dagood) [reply src/pkg/projects/netcoreapp/pkg/Microsoft.NETCore.App.Runtime.pkgproj]: We'd need a workaround to get it into the sharedfx after the move, but I think it's more consistent that way at least. I'll file an issue.

File: src/pkg/packaging-tools/framework.dependency.targets [hunk 3]
```diff
@@ -62,6 +62,25 @@
...
+    Without this, the irrelevant properties cause a cache miss. Both restores run at the same time
+    with the same intermediate location, causing file lock errors.
+  -->
+  <Target Name="ParallelSafeRestore"
+          BeforeTargets="RestorePackages"
+          Condition="'$(InParallelSafeRestore)' != 'true'">
+    <MSBuild
+      Projects="$(MSBuildProjectFullPath)"
+      Targets="RestorePackages"
+      RemoveProperties="BaseId;IdPrefix"
```
[the remark this answers is not in the document]
Comment (dagood) [reply src/pkg/packaging-tools/framework.dependency.targets]: Back when I implemented suffixed RIDs for the apphost pack I looked into it, but it didn't seem reasonable without making changes to the packaging tools. On that note, filed https://github.com/dotnet/arcade/issues/2997!

I do have https://github.com/dotnet/core-setup/issues/6696 tracking removing this workaround in general, with some other ways foward.

File: src/pkg/projects/netcoreapp/src/netcoreapp.depproj (hunk 1 above)
Comment (dagood) [reply src/pkg/projects/netcoreapp/src/netcoreapp.depproj]: True, filed https://github.com/dotnet/arcade/issues/2998.

File: src/pkg/projects/netcoreapp/pkg/Microsoft.NETCore.App.Runtime.pkgproj (hunk 2 above)
[the remark this answers is not in the document]
Comment (ericstj) [reply src/pkg/projects/netcoreapp/pkg/Microsoft.NETCore.App.Runtime.pkgproj]: Ok, make sure that this code will fail once they move it so you don't have to remember to clean it up.

[the remark this answers is not in the document]
Comment (dagood) [reply src/pkg/projects/netcoreapp/pkg/Microsoft.NETCore.App.Runtime.pkgproj]: It turns out `createdump` is included in SCD, and looking back, my checkbox for changing that doesn't have any grounding. Removing this target. Big thanks for questioning it. 😄

**Pull request merged**